pymdownx/snippets.py

```diff
@@ -298,26 +300,27 @@ def parse_snippets(self, lines, file_name=None, is_url=False, is_section=False):
                     continue
 
                 # Get line numbers (if specified)
-                end = None
-                start = None
+                end = []
+                start = []
                 section = None
                 m = self.RE_SNIPPET_FILE.match(path)
-                path = m.group(1).strip()
+                path = '' if m is None else m.group(1).strip()
                 # Looks like we have an empty file and only lines specified
                 if not path:
                     if self.check_paths:
                         raise SnippetMissingError(f"Snippet at path '{path}' could not be found")
                     else:
                         continue
-                ending = m.group(3)
-                if ending and len(ending) > 1:
-                    end = int(ending[1:])
-                starting = m.group(2)
-                if starting and len(starting) > 1:
-                    start = max(0, int(starting[1:]) - 1)
-                section_name = m.group(4)
-                if section_name:
-                    section = section_name[1:]
+                if m.group(2):
+                    for nums in m.group(2)[1:].split(','):
+                        span = nums.split(':')
+                        start.append(max(0, int(span[0]) - 1) if span[0] else None)
+                        if len(span) > 1:
+                            end.append(int(span[1]) if span[1] else None)
+                        else:
+                            end.append(None)
```

Comment:
Sorry, I missed that: Another candidate for possible simplification (redundancy reduction):
```suggestion
                        if len(span) > 1 and span[1]:
                            end.append(int(span[1]))
                        else:
                            end.append(None)
```
or
```suggestion
                        end.append(int(span[1]) if len(span) > 1 and span[1] else None)
```